### ISSUE TITLE
Problems with Includes

### ISSUE THREAD

**Issue by killercup** [walling/gulp-raml2html#6]
Hi,

I just wanted to try this out and got some errors concerning missing files for includes.

Gulpfile:

``` coffee
gulp.task 'api_docs', ->
  gulp.src('api/*.raml')
  .pipe raml2html()
  .pipe gulp.dest('public/api')
```

RAML file:

``` yaml
schemas:
  - show: !include schemas-show.json
```

```
Error: /Users/pascal/Projekte/EF/public/api/ef-api.raml:27:11: Parse error while reading schemas-show.json: cannot read schemas-show.json (Error: ENOENT, open 'schemas-show.json')
  at /Users/pascal/Projekte/EF/node_modules/gulp-raml2html/index.js:32:16
  at process._tickCallback (node.js:419:13)
```

From what I can tell, this is because [here](https://github.com/walling/gulp-raml2html/blob/91e34783d36674b6cc3bcfeb3f2b359002b94f4d/index.js#L67) the first action of the plugin is pushing the `file` object back into the stream. But when [convertFile](https://github.com/walling/gulp-raml2html/blob/91e34783d36674b6cc3bcfeb3f2b359002b94f4d/index.js#L38) is later called, the file's path has been changed by `gulp.dest`(cf. [this](https://github.com/wearefractal/vinyl/blob/80d3f61445b347fc1c34f462f0ab800644e90e04/index.js#L162)).

Long story short, I moved the `this.push(file);` line to the end of the function and it seems to work.

(Maybe I'll send a pull request later, I just wanted to write this down before I forget.)

**Comment by rdohms** [comment]
I'm still having issues with included schemas and resolving paths, any news on a fix?

**Comment by nvh** [comment]
I've already created [this pull request](https://github.com/walling/gulp-raml2html/pull/8), but it isn't merged yet

**Comment by iki** [comment]
yep, I can confirm that includes don't work if dest dir != source dir, and also the dest dir is not created if it does not exist

used 3-step workaround for now:
- `apidoc-html` - generate html in source dir
- `apidoc-copy` - copy html to dest dir
- `apidoc` - remove html in source dir
```js
var del = require('del');
var path = require('path');
var gulp = require('gulp');
var rename = require('gulp-rename');
var raml2html = require('gulp-raml2html');

var RAML   = '../api/api.raml';
var APIDOC = '../server/static/docs/api/index.html';

gulp.task('apidoc', ['apidoc-copy'], function(done) {
  del(RAML.replace(/\.raml$/, '.html'), {force: true}, done);
});

gulp.task('apidoc-copy', ['apidoc-html'], function() {
  return gulp.src(RAML.replace(/\.raml$/, '.html'))
    .pipe(rename(path.basename(APIDOC)))
    .pipe(gulp.dest(path.dirname(APIDOC)));
});

gulp.task('apidoc-html', function() {
  return gulp.src(RAML)
    .pipe(raml2html())
    .pipe(gulp.dest(path.dirname(RAML)));
});
```

**Comment by iki** [comment]
another workaround is to just use current raml2html directly:
https://gist.github.com/iki/784ddd5ab33c1e1b726b
... which is the only workaround for #12 atm
